Classes/common/CDTDatastore.m

```diff
@@ -63,6 +64,7 @@ - (id)initWithDatabase:(TD_Database *)database
         NSString *name = [database name];
         _extensionsDir =
             [dir stringByAppendingPathComponent:[NSString stringWithFormat:@"%@_extensions", name]];
+        _manager = manager;
```

Comment:
We should probably get better at validating pre-conditions, and documenting them. At least document if a parameter is required, but asserting is better (particularly as the macros are compiled out in production builds if I recall).

Perhaps [NSParameterAssert](https://developer.apple.com/library/mac/documentation/Cocoa/Reference/Foundation/Miscellaneous/Foundation_Functions/#//apple_ref/c/macro)./NSParameterAssert